lightning/src/chain/keysinterface.rs

```diff
@@ -139,13 +139,20 @@ pub trait ChannelKeys : Send {
 	///
 	/// Note that if signing fails or is rejected, the channel will be force-closed.
 	///
+	/// The commitment_tx follows BIP-69 lexicographical ordering.
+	///
+	/// The redeem_scripts vector is 1-1 mapped to commitment_tx outputs.  For p2wpkh, the
```

Comment:
Then it should probably be a vec containing pairs, not two vecs.

Reply:
The outputs vec is buried inside the `Transaction` object, so I assume this has to wait until we move to tx construction in the signer.

Reply:
Unless you're thinking of moving the redeem script into `Transaction`?